cartridges/int_algolia/cartridge/scripts/algolia/helper/reindexHelper.js

```diff
@@ -0,0 +1,108 @@
+const logger = require('dw/system/Logger').getLogger('algolia');
+
+var algoliaData = require('*/cartridge/scripts/algolia/lib/algoliaData');
+var algoliaIndexingAPI = require('*/cartridge/scripts/algoliaIndexingAPI');
+
+/**
+ * Delete the temporary indices corresponding to the given type and locales
+ * @param {string} indexType - type of the index (products or categories)
+ * @param {string[]} locales - locales for which we want to delete the indices
+ * @return {Object} Algolia taskIDs, in the form: { indexName1: <taskID>, indexName2: <taskID> }
+ */
+function deleteTemporariesIndices(indexType, locales) {
+    var deletionTasks = {};
+    locales.forEach(function(locale) {
+        var tmpIndexName = algoliaData.calculateIndexName(indexType, locale) + '.tmp';
+        var res = algoliaIndexingAPI.deleteIndex(tmpIndexName);
+        if (res.ok) {
+            logger.info(tmpIndexName + ' deleted. ' + JSON.stringify(res.object.body));
+            deletionTasks[tmpIndexName] = res.object.body.taskID;
+        } else {
+            throw new Error('Error while deleting temporary indices: ' + res.getErrorMessage())
+        }
+    });
+    return deletionTasks;
+}
+
+/**
+ * Copy the settings of the production indices to the matching temporary index
+ * @param {string} indexType - type of the index (products or categories)
+ * @param {string[]} locales - locales for which we want to copy the index settings
+ * @return {Object} Algolia taskIDs, in the form: { indexName1: <taskID>, indexName2: <taskID> }
+ */
+function copySettingsFromProdIndices(indexType, locales) {
```

Comment:
Does this also include replicas?

Reply:
Replicas have their own settings which won't be touched. What happens is that when the temporary index is moved to production, the replica will only synchronize its data